python/activator/middleware_interface.py

```diff
@@ -1760,23 +1754,32 @@ def _filter_datasets(src_repo: Butler,
         ", ".join(repr(a) for a in args),
         ", ".join(f"{k}={v!r}" for k, v in kwargs.items()),
     )
-    try:
-        with lsst.utils.timer.time_this(_log, msg=f"_filter_datasets({formatted_args}) (known datasets)",
-                                        level=logging.DEBUG):
-            known_datasets = set(dest_repo.registry.queryDatasets(*args, **kwargs))
+    # time_this automatically escalates its log to ERROR level if the code raises.
+    # Some errors are expected to happen sometimes, and we don't want those to log
+    # at the ERROR level and cause confusion.
+    with lsst.utils.timer.time_this(_log, msg=f"_filter_datasets({formatted_args}) (known datasets)",
+                                    level=logging.DEBUG):
+        try:
+            # Okay to have empty results.
+            known_datasets = set(dest_repo.query_datasets(explain=False, *args, **kwargs))
             _log_trace.debug("Known datasets: %s", known_datasets)
-    except lsst.daf.butler.registry.DataIdValueError as e:
-        _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
-        # If dimensions are invalid, then *any* such datasets are missing.
-        known_datasets = set()
+        except MissingDatasetTypeError as e:
+            _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
+            known_datasets = set()
+        except DataIdValueError as e:
+            _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
+            # If dimensions are invalid, then *any* such datasets are missing.
+            known_datasets = set()
 
     # Let exceptions from src_repo query raise: if it fails, that invalidates
     # this operation.
     # "expanded" dimension records are ignored for DataCoordinate equality
     # comparison, so we only need them on src_datasets.
     with lsst.utils.timer.time_this(_log, msg=f"_filter_datasets({formatted_args}) (source datasets)",
                                     level=logging.DEBUG):
-        src_datasets = set(src_repo.registry.queryDatasets(*args, **kwargs).expanded())
+        # Ok with empty query result here, not log an error, and let the downstream
+        # method decide what to do with empty results.
```

Comment:
When I wrote that, I meant the `EmptyQueryResultError` raise by butler and `explain=False` tells butler to not raise for empty results.  My comments weren't clear and I'm re-wording my comments. 

I agree with your second comment. I am worried about the `_MissingDatasetError` in this code versus butler's `EmptyQueryResultError`. I think it might be a combination of DM-40245 and DM-46178.  I'm adding that to the comment.